_posts/2019/2019-03-28-be-aware-of-nil-values-when-using-where-not.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Be Aware of nil values when using where.not()
+description: Strange Think
+layout: article
+category: ruby
+image:
+  base: '2019/be-aware-of-nil-values-when-using-where-not'
+  alt: 'Half-finished painted wall'
+  source: "https://unsplash.com/photos/oBb-Y26PJgg"
+  credit: "Samuel Zeller"
+---
+
+The use of `where.not` when building Active Record scopes can help you to elegantly specify scopes where an attribute does not have a specific value.
+
+However the abstraction has some quirks if you have fields that can be `NULL` in your database (`nil` in Ruby).
+
+
+## Instead of…
+
+…expecting `where.not` to return `nil` values:
+
+```ruby
+non_soy_options = Coffee.where.not(milk: "Soy")
+```
+
+
+## Use…
+
+…an `or` clause to explicitly request the `nil` values.
```

Comment:
Oh! So I think you want to say at the beginning maybe more explicitly that it doesn't return 'nil' columns by default and there are cases where you may want/expect that.